app/models/propertyrepresentation/AgentOrganisation.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.propertyrepresentation
+
+import java.time.LocalDateTime
+
+import play.api.libs.json.{Json, OFormat}
+
+case class AgentOrganisation(
+      id: Long,
+      representativeCode: Option[Long],
```

Comment:
if this is an AGENT account, the representative code should be mandatory.

Reply:
representativeCode is not always returned by the endpoint (according to the schema).